app/models/katello/host/subscription_facet.rb

```diff
@@ -50,9 +50,13 @@ def update_hypervisor(consumer_params)
         end
       end
 
-      def update_guests
+      def update_guests(consumer_params)
         if self.hypervisor
-          guests = self.candlepin_consumer.virtual_guests
+          if !consumer_params.try(:[], 'guestIds').empty?
+            guests = FactValue.where("value IN (?)", consumer_params['guestIds']).pluck('host_id')
```

Comment:
this line should find fact values only where fact name is "virt::uuid" (I think that is the syntax for candlepin facts). I wasn't sure how best to do that.